classes/Address.php

```diff
@@ -234,9 +234,13 @@ public function delete()
 
             return parent::delete();
         } else {
-            $this->deleted = true;
-
-            return $this->update();
+            return Db::getInstance()->update(
```

Comment:
Hello, I'm afraid I'm not sure to understand what is the goal here.

My guess is... if you expect `$this->update()` to fail, you modify the database directly instead?

But there might be some logic in `$this->update()` that will not be executed in that case.